compiler/test-resources/repl/settings-repl-max-print-elements

```diff
@@ -0,0 +1,7 @@
+scala> 1.to(300).toList
+val res0: List[Int] = List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 22 ... large output truncated, print value to show all
+
+scala>:settings -Vrepl-max-print-elements:20
+
+scala> 1.to(300).toList
+val res1: List[Int] = List(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20)
```

Comment:
We should somehow indicate that the printed output was truncated, similarly to how it's done when the limit of characters gets exceeded

Reply:
That would be nice, but [`ScalaRunTime.stringOf`](https://github.com/scala/scala/blob/fa51ca3/src/library/scala/runtime/ScalaRunTime.scala#L187) only returns the resulting String...

I can see a few ways to achieve that, but none are particularly nice:
1) we could invoke it twice - once with the limit, once without. If the string.length differs, we append the info that it was truncated. Downside: slower, and traversing the result may have side effects, dependent on what `stringOf` actually does. 

2) fork/copy the `stringOf` implementation and indicate if it was truncated in the result. Downside: duplicating code that looks rather complex. 

3) like 2), but bring this into scala2-library
Downside: need to wait until the next scala2 release before we can use it. 

Thoughts?

Reply:
I would go with 1). As REPL is used mainly for experimenting, I wouldn't expect extremely big outputs very often, so the efficiency of printing should not be a very big problem. On the other hand, getting wrong string representation might bring people to wasting a lot of time trying to figure out why the output was not what they might have expected. And if `stringOf` could have any side effects, that would mean we should not use it for printing things in REPL at all.

Reply:
done - I also added in some minor refactorings, some comments and updated the test

Reply:
This is the Scala 2 side PR that adds ellipsis. Maybe I can get that piece accepted quickly (2.13.10 anyone?).

https://github.com/scala/scala/pull/8885/files

The whole point of that PR was to support Scala 3 nicely. (Not supplying extra newlines, etc.)